modules/plugin/http-commons/src/main/java/org/geotools/http/commons/MultithreadedHttpClient.java

```diff
@@ -77,123 +78,83 @@
+                        .build();
 
         client = createHttpClient();
-
-        applySystemProxySettings();
     }
 
+    HttpClientBuilder clientBuilder = HttpClientBuilder.create();
     // package private to support testing || JNH Public for refactoring
     public HttpClient createHttpClient() {
-        return new HttpClient(connectionManager);
-    }
-
-    private void applySystemProxySettings() {
-        final String proxyHost = System.getProperty("http.proxyHost");
-        final int proxyPort = Integer.parseInt(System.getProperty("http.proxyPort", "80"));
-        String nonProxyHostProp = System.getProperty("http.nonProxyHosts");
-
-        if (proxyHost != null) {
-            LOGGER.fine(
-                    "Found 'http.proxyHost' Java System property. Using it as proxy server. Port: "
-                            + proxyPort);
-            HostConfiguration hostConfig = client.getHostConfiguration();
-            if (nonProxyHostProp != null) {
-                if (nonProxyHostProp.startsWith("\"")) {
-                    nonProxyHostProp = nonProxyHostProp.substring(1);
-                }
-                if (nonProxyHostProp.endsWith("\"")) {
-                    nonProxyHostProp = nonProxyHostProp.substring(0, nonProxyHostProp.length() - 1);
-                }
-                hostConfigNoProxy = (HostConfiguration) hostConfig.clone();
-                StringTokenizer tokenizer = new StringTokenizer(nonProxyHostProp, "|");
-                while (tokenizer.hasMoreTokens()) {
-                    nonProxyHosts.add(tokenizer.nextToken().trim().toLowerCase());
-                }
-                LOGGER.fine("Initialized with nonProxyHosts: " + nonProxyHosts);
-            }
-            hostConfig.setProxy(proxyHost, proxyPort);
-        }
-
-        final String proxyUser = System.getProperty("http.proxyUser");
-        final String proxyPassword = System.getProperty("http.proxyPassword");
-        if (proxyUser != null) {
-            if (proxyPassword == null || proxyPassword.length() == 0) {
-                LOGGER.warning(
-                        "System property http.proxyUser provided but http.proxyPassword "
-                                + "not provided or empty. Proxy auth credentials will be passed as is anyway.");
-            } else {
-                LOGGER.fine(
-                        "System property http.proxyUser and http.proxyPassword found,"
-                                + " setting proxy auth credentials");
-            }
-            HttpState state = client.getState();
-            if (state == null) {
-                state = new HttpState();
-                client.setState(state);
-            }
-            AuthScope authscope = AuthScope.ANY;
-            Credentials credentials = new UsernamePasswordCredentials(proxyUser, proxyPassword);
-            state.setProxyCredentials(authscope, credentials);
-        }
+        clientBuilder.useSystemProperties();
```

Comment:
Oh neat... and it accounts for even more system properties than the old code did, see the [class javadoc](https://hc.apache.org/httpcomponents-client-4.5.x/current/httpclient/apidocs/index.html?org/apache/http/impl/client/HttpClientBuilder.html) (it's not in the method javadoc for some reason).

Reply:
Actually, we should probably add that list to our docs somewhere